bottleneck/src/reduce_template.c

```diff
@@ -1290,41 +1292,49 @@ reducer(char *name,
     if (reduce_all == 1) {
         /* we are reducing the array along all axes */
         if (dtype == NPY_FLOAT64) {
-            return fall_float64(a, ddof);
+            y = fall_float64(a, ddof);
         }
         else if (dtype == NPY_FLOAT32) {
-            return fall_float32(a, ddof);
+            y = fall_float32(a, ddof);
         }
         else if (dtype == NPY_INT64) {
-            return fall_int64(a, ddof);
+            y = fall_int64(a, ddof);
         }
         else if (dtype == NPY_INT32) {
-            return fall_int32(a, ddof);
+            y = fall_int32(a, ddof);
         }
         else {
-            return slow(name, args, kwds);
+            y = slow(name, args, kwds);
         }
     }
     else {
         /* we are reducing an array with ndim > 1 over a single axis */
         if (dtype == NPY_FLOAT64) {
-            return fone_float64(a, axis, ddof);
+            y = fone_float64(a, axis, ddof);
         }
         else if (dtype == NPY_FLOAT32) {
-            return fone_float32(a, axis, ddof);
+            y = fone_float32(a, axis, ddof);
         }
         else if (dtype == NPY_INT64) {
-            return fone_int64(a, axis, ddof);
+            y = fone_int64(a, axis, ddof);
         }
         else if (dtype == NPY_INT32) {
-            return fone_int32(a, axis, ddof);
+            y = fone_int32(a, axis, ddof);
         }
         else {
-            return slow(name, args, kwds);
+            y = slow(name, args, kwds);
         }
 
     }
 
+    Py_DECREF(a);
+
+    return y;
+
+    cleanup:
```

Comment:
nit: this is typically dedented (at least in the NumPy code I've seen). Also if you're only using it for error handling, I might call it `error` instead.